problem_builder/table.py

```diff
@@ -66,23 +74,122 @@ class MentoringTableBlock(StudioEditableXBlockMixin, StudioContainerXBlockMixin,
+                # Use default settings.
+                pass
+            else:
+                raise JsonHandlerError(403, _("You are not permitted to view this student's table."))
+
         for child_id in self.children:
             child = self.runtime.get_block(child_id)
             # Child should be an instance of MentoringTableColumn
             header_values.append(child.header)
             child_frag = child.render('mentoring_view', context)
             content_values.append(child_frag.content)
-            fragment.add_frag_resources(child_frag)
         context['header_values'] = header_values if any(header_values) else None
         context['content_values'] = content_values
+        html = loader.render_template('templates/html/mentoring-table.html', context)
+        return {'content': html}
+
+    @property
+    def current_user_key(self):
+        user = self.runtime.service(self, 'user').get_current_user()
+        # We may be in the SDK, in which case the username may not really be available.
+        return user.opt_attrs.get('edx-platform.username', 'username').lower()
+
+    @XBlock.json_handler
+    def get_shared_list(self, data, suffix=''):
+        context = {'shared_with': self.shared_with.get(self.current_user_key, [])}
+        return {
+            'content': loader.render_template('templates/html/mentoring-table-shared-list.html', context)
+        }
+
+    @XBlock.json_handler
+    def share_results(self, data, suffix=''):
+        target_usernames = data.get('usernames')
+        target_usernames = [username.strip().lower() for username in target_usernames if username.strip()]
+        current_username = self.current_user_key
+        if not target_usernames:
+            raise JsonHandlerError(400, _('Usernames not provided.'))
+        shared_with = self.shared_with.get(current_username, [])
+        for target_username in target_usernames:
+            if not target_username:
+                continue
+            view_options = self.view_options.get(target_username, [])
+            if current_username not in view_options:
+                view_options.append(current_username)
+            if target_username not in shared_with:
+                shared_with.append(target_username)
+            self.view_options[target_username] = view_options
+        self.shared_with[current_username] = shared_with
+        return {'message': _('Shared with {username}.').format(username=target_username)}
+
+    @XBlock.json_handler
+    def remove_share(self, data, suffix=''):
+        target_username = data.get('username')
+        if not target_username:
+            raise JsonHandlerError(400, _('Username not provided.'))
+        current_username = self.current_user_key
+        view_options = self.view_options.get(target_username, [])
+        shared_with = self.shared_with.get(current_username, [])
+        view_options = [option for option in view_options if option != current_username]
```

Comment:
@Kelketek if view_options and shared_with were sets it would be enough to do `view_options -= {current_username}`